runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -938,7 +947,21 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
 
    // Put the request back into the pool
    setMethodBeingCompiled(NULL);
-   compInfo->requeueOutOfProcessEntry(&entry);
+
+   if (!compInfo->getPersistentInfo()->getDisableFurtherCompilation()
+       && !enableJITServerPerCompConn)
+      {
+      compInfo->requeueOutOfProcessEntry(&entry);
+      }
+   else
+      {
+      // Delete server stream if per compilation connections are enabled
+      // or if the server disabled compilations due to a crash
+      stream->~ServerStream();
```

Comment:
Note that on line 921 we call the destructor of the stream as well. To be safe we may want to check for `stream != NULL` or groups the two similar sequences together

Reply:
Yes, looking at the code, a bunch of other calls to `ServerStream` destructor can be removed from exception handlers as they should all be handled in the `abortCompilation` if statement.